x-pack/plugins/security_solution/server/lib/telemetry/index.ts

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { UsageCollectionSetup } from 'src/plugins/usage_collection/server';
+import { ISavedObjectsRepository } from '../../../../../../src/core/server';
+import {
+  EndpointMetadataTelemetry,
+  getDefaultEndpointMetadataTelemetry,
+  getEndpointMetadataTelemetryFromFleet,
+} from './endpoint_metadata_telemetry';
+
+const SIEM_USAGE_COLLECTOR_TYPE = 'security_solution';
+export async function createSiemTelemetry(
+  usageCollector: UsageCollectionSetup,
+  savedObjectsClient: ISavedObjectsRepository
+) {
+  const collector = usageCollector.makeUsageCollector({
+    type: SIEM_USAGE_COLLECTOR_TYPE,
+    isReady: () => true,
+    fetch: async () => {
+      try {
+        const endpointMetadataTelemetryFromFleet: EndpointMetadataTelemetry = await getEndpointMetadataTelemetryFromFleet(
+          savedObjectsClient
+        );
+        return {
+          ...endpointMetadataTelemetryFromFleet,
+        };
+      } catch (err) {
+        if (err.output?.statusCode === 404) {
+          return {};
+        }
+        return getDefaultEndpointMetadataTelemetry(); // TODO: no data returned if an error occured or should fail silently here?
```

Comment:
Hmm I wonder if we should return `{}` if an error occurs because we don't know the results? It's probably not very helpful to send 0s right?

Reply:
Nope, I was back and forth on this and saw different implementations, but I agree that an empty object is definitely more informative.